plugins/org.eclipse.fordiac.ide.gef/src/org/eclipse/fordiac/ide/gef/DiagramEditor.java

```diff
@@ -173,7 +177,17 @@ protected Point getInitialScrollPos(final GraphicalEditPart rootEditPart) {
 	protected void createGraphicalViewer(final Composite parent) {
 		final RulerComposite rulerComp = new FordiacRulerComposite(parent, SWT.NONE);
 
-		final var prefCache = new GefPreferenceConstantsCache(getSystem().getTypeLibrary().getProject());
+		IProject project = null;
+		final IEditorInput input = getEditorInput();
+		if (input instanceof final IContentEditorInput contentInput) {
+			if (EcoreUtil.getRootContainer(contentInput.getContent()) instanceof final LibraryElement libElement) {
```

Comment:
You can use `TypeLibraryManager.INSTANCE.getTypeLibraryFromContext(contentInput.getContent()).getProject()` to reduce code duplication.